### PR TITLE
fix: migrate logging to slf4j

### PR DESCRIPTION
**Pull request by bjagg**
Log4j has seen some security issues lately. While this codebase did not use log4j, it had been included as dependency (likely from an early time).

Removing Log4j and Jakarta Commons Logging and replacing with SLF4J + Logback, which is the uPortal preferred logger.

### CODE REVIEW

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  <!--
+    | Expose the logback configuration via JMX
+  +-->
+  <jmxConfigurator />
+
+  <!--
+    | Specify a local property that sets up the logging directory (to tomcat's logs directory).
+  +-->
+  <property scope="local" name="LOG_DIR" value="${bookmarks.root}/../../logs" />
```
Comment (jonathanmtran) [inline src/main/resources/logback.xml:47]: Should `LOG_DIR` be set/configured like https://github.com/uPortal-Project/NotificationPortlet/blob/master/notification-portlet-webapp/src/main/resources/logback.xml#L48-L53 ?

Comment (bjagg) [reply src/main/resources/logback.xml]: Yeah, let's make it the default to expect the Tomcat layout.

File: src/main/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    </rollingPolicy>
+  </appender>
+
+  <!--
+    | Setup default log level to INFO
+  +-->
+  <root level="INFO">
+    <appender-ref ref="LOG" />
+  </root>
+
```
[the remark this answers is not in the document]
Comment (bjagg) [reply src/main/resources/logback.xml:88]: Nice!

[the remark this answers is not in the document]
Comment (bjagg) [reply src/main/resources/logback.xml:88]: Appreciate you taking the time to review in detail @jonathanmtran

**Review by jonathanmtran — APPROVED** [review]

**Review by jgribonvald — APPROVED** [review]

**Review by ChristianMurphy — APPROVED** [review]

**Pull request merged**